sdk/core/azure-core/src/main/java/com/azure/core/credential/TokenCredential.java

```diff
@@ -5,15 +5,31 @@
 
 import reactor.core.publisher.Mono;
 
+import java.time.Duration;
+
 /**
  * The interface for credentials that can provide a token.
  */
 @FunctionalInterface
 public interface TokenCredential {
+    /**
+     * The default duration before the actual token expiry to refresh the token. The value is 2 minutes.
+     */
+    Duration DEFAULT_TOKEN_REFRESH_OFFSET = Duration.ofMinutes(2);
+
     /**
      * Asynchronously get a token for a given resource/audience.
      * @param request the details of the token request
      * @return a Publisher that emits a single access token
      */
     Mono<AccessToken> getToken(TokenRequestContext request);
+
+    /**
+     * The duration before the actual token expiry to refresh the token. Default is 2 minutes.
+     *
+     * @return the current offset for token refresh
```

Comment:
This is a little hard to parse. Maybe something like "Returns a Duration value representing the amount of time to subtract from the token expiry time, whereupon attempts will be made to refresh the token. By default this will occur two minutes prior to the expiry of the token."

Reply:
This is moved to `TokenRefreshOptions` and the javadocs are updated there.